### PR TITLE
daemon/api_quotas.go: handle conflicts, returning conflict response

### PR DESCRIPTION
**Pull request by anonymouse64**
Use a more proper error kind for quota conflicts, supporting the error type
*servicestate.QuotaChangeConflictError inside errToResponse and indicating what
sort of change the conflict is being used for.

Also update the unit tests for the daemon error response to check for this.

### CODE REVIEW

**Comment by mardy** [comment]
Now that the other branch is merged, maybe you can merge the master branch into this one, in order to reduce the diff.

File: daemon/api_quotas.go [hunk 1]
```diff
@@ -176,7 +177,14 @@ func postQuotaGroup(c *Command, r *http.Request, _ *auth.UserState) Response {
 			// then we need to create the quota
 			ts, err = servicestateCreateQuota(st, data.GroupName, data.Parent, data.Snaps, quantity.Size(data.MaxMemory))
 			if err != nil {
-				// XXX: dedicated error type?
+				// check for conflict errors
+				switch err.(type) {
+				case *snapstate.ChangeConflictError:
+					return Conflict(err.Error())
```
Comment (pedronis) [inline daemon/api_quotas.go:183]: we want to use errToResponse here or SnapChangeConflict

File: daemon/api_quotas.go [hunk 2]
```diff
@@ -176,7 +177,14 @@ func postQuotaGroup(c *Command, r *http.Request, _ *auth.UserState) Response {
...
+				case *servicestate.QuotaChangeConflictError:
+					return Conflict(err.Error())
```
Comment (pedronis) [inline daemon/api_quotas.go:185]: same, we want to introduce a client.ErrorKindQuotaChangeConflict and have similar level of support, so it should be set together with carrying the quota-name and the change-kind

**Review by pedronis — CHANGES_REQUESTED** [review]
this needs more work to have the typical level of support we have for snaps conflicts

File: daemon/api_quotas.go (hunk 1 above)
Comment (anonymouse64) [reply daemon/api_quotas.go]: done

File: daemon/api_quotas.go (hunk 2 above)
Comment (anonymouse64) [reply daemon/api_quotas.go]: done

**Review by bboozzoo — APPROVED** [review]
LGTM

**Comment by mvo5** [comment]
Unit tests fail with:
```
2021-07-05T18:46:34.2718351Z There are files left in the git tree after the tests:
2021-07-05T18:46:34.2718741Z 
2021-07-05T18:46:34.2720066Z ?? "cmd/snap-seccomp/A\016\360\001Z\016\300\001\003\031\001\016\310\001Q\016\320\001S\016\330\001D\016\340\001D\016\350\001A\016\360\001b\016\300\001[\n\0168A\0160A\016(B\016 B\016\030B\016\020B\016\bE\v"
2021-07-05T18:46:34.2722477Z 
```
I have never seen this failure before, could be random FS corruption of course.

**Comment by bboozzoo** [comment]
> Unit tests fail with:
> 
> ```
> 2021-07-05T18:46:34.2718351Z There are files left in the git tree after the tests:
> 2021-07-05T18:46:34.2718741Z 
> 2021-07-05T18:46:34.2720066Z ?? "cmd/snap-seccomp/A\016\360\001Z\016\300\001\003\031\001\016\310\001Q\016\320\001S\016\330\001D\016\340\001D\016\350\001A\016\360\001b\016\300\001[\n\0168A\0160A\016(B\016 B\016\030B\016\020B\016\bE\v"
> 2021-07-05T18:46:34.2722477Z 
> ```
> 
> I have never seen this failure before, could be random FS corruption of course (and given that this PR changed quite different things highly unlikely that it has anything to do with it :)

Looks like some random garbage in fs? I've restarted the github job.

**Review by pedronis — APPROVED** [review]
thank you for the refinement

**Pull request merged**